crates/parser/src/ix_parser/token_program/token_program_parser.rs

```diff
@@ -0,0 +1,311 @@
+use spl_pod::solana_program::{program_error::ProgramError, pubkey::Pubkey};
+use spl_token::instruction::TokenInstruction;
+use yellowstone_grpc_proto::prelude::Transaction;
+use yellowstone_vixen_core::{ParseResult, Parser, Prefilter, TransactionUpdate};
+
+use super::token_ix_context::*;
+use crate::ix_parser::vixen_ix::structure::{
+    Instruction, InstructionData, InstructionsUpdate, ReadableInstructions,
+};
+
+#[derive(Debug)]
+pub struct TokenProgramIxsParser;
+
+pub type TokenProgramIxs = Vec<ReadableInstructions<TokenIxContext>>;
+
+impl Instruction<TokenIxContext> for TokenProgramIxsParser {
+    fn get_ix_context(ix_data: &InstructionData) -> Result<TokenIxContext, ProgramError> {
+        let ix_type = TokenInstruction::unpack(&ix_data.data)?;
+
+        match ix_type {
+            TokenInstruction::Transfer { amount } => {
+                let source = ix_data.accounts[0];
+                let destination = ix_data.accounts[1];
+                Ok(TokenIxContext::Transfer(Transfer {
+                    source,
+                    destination,
+                    amount,
+                }))
+            },
```

Comment:
Accounts should be in another struct. We have TokenInstruction as the data and then accounts being a context struct with the accounts.

Reply:
![image](https://github.com/user-attachments/assets/0f843fbd-2c95-4938-b5c4-a38d9ff135ee)
something like this?